module/util/log.go

```diff
@@ -4,18 +4,24 @@ import (
 	"github.com/rs/zerolog"
 )
 
+// LogProgress takes a total and return function such that when called with a 0-based index
+// it prints the progress from 0% to 100% to indicate the index from 0 to (total - 1) has been
+// processed.
+// useful to report the progress of processing the index from 0 to (total - 1)
 func LogProgress(msg string, total int, logger *zerolog.Logger) func(currentIndex int) {
-	logThreshold := float64(10)
+	lookup := make(map[int]int)
```

Comment:
The existing implementation tries to check if the current index has reached a percentage that parsed a threshold (such as 10%) to trigger the logging. 

This creates some problem that when total is small, we end up logging as `11%` or `12%`, instead of `10%`, although the `11%`.

The current approach does the other around, it builds a lookup to determine which index would trigger the 10%, 20%, ... 100%. and during iteration, we use the current index to find from the lookup. This implementation would be easier to think IMO

Reply:
we has this discussion before.  the problem with using the precomputed map is that your total may be inaccurate / an estimate, and you stop logging after 100% (even if it in reality goes beyond 100%)

you can just round down the logging number to the nearest unit (or just use the logThreshold)?